File: fieldmate/src/main/java/com/nguyencong/fieldmate/service/BenefitService.java
package com.nguyencong.fieldmate.service;

import com.nguyencong.fieldmate.dto.request.BenefitRequest;
import com.nguyencong.fieldmate.dto.response.BenefitResponse;

public interface BenefitService {
    BenefitResponse createBenefit(Long venueId, BenefitRequest request);
    BenefitResponse updateBenefit(Long id, BenefitRequest request);
    void deleteBenefit(Long id);
}

File: front-end/fieldmate-web/src/services/benefit.service.ts
import {
  fieldmateClient,
  fieldmateEndpoints,
} from "@/configs/fieldmate-client";
import type {
  BenefitRequest,
  BenefitResponse,
} from "@/types/benefit";

export const benefitService = {
  async create(venueId: number, data: BenefitRequest) {
    const response = await fieldmateClient.post<BenefitResponse>(
      fieldmateEndpoints.secureVenueBenefits(venueId),
      data,
    );

    return response.data;
  },

  async update(benefitId: number, data: BenefitRequest) {
    const response = await fieldmateClient.put<BenefitResponse>(
      fieldmateEndpoints.secureBenefit(benefitId),
      data,
    );

    return response.data;
  },

  async remove(benefitId: number) {
    await fieldmateClient.delete(
      fieldmateEndpoints.secureBenefit(benefitId),
    );
  },
};
